.circleci/config.yml

```diff
@@ -25,18 +25,30 @@ jobs:
     machine: true
     steps:
       - checkout
+      - setup_remote_docker
       - run:
           command: |
             docker build -t lubot .
+            docker save -o /caches/app.tar lubot
+      - save_cache:
+          key: v1-{{ .Branch }}-{{ epoch }}
```

Comment:
You could use `v1-{{ .Branch }}-{{ checksum "/caches/app.tar" }}` so the cache only gets refreshed when there are changes to the tar file in a given branch.